tests/workflow/test_construct_batch.py

```diff
@@ -368,8 +371,21 @@ def circuit(order):
 
         assert fn((1.0, 2.0)) == ((1.0, 2.0),)
 
+    def test_level_none_deprecated(self):
+        """Test that level=None raises a deprecation warning."""
+
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit():
+            return qml.state()
+
+        with pytest.warns(
+            PennyLaneDeprecationWarning,
+            match="`level=None` is deprecated",
+        ):
+            get_transform_program(circuit, level=None)
```

Comment:
Can we switch this test to one of `construct_batch`? Especially because `get_transform_program` is kinda inaccurate these days so we need to consider getting rid of it.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7886/commits/c58cf6740e029e2522e7837f3246e09e7390a4e8